.tito/lib/git_annex_spec_builder.py

```diff
@@ -0,0 +1,168 @@
+        our build tag was created.
+
+        Created in the temporary rpmbuild SOURCES directory.
+        """
+        self._create_build_dirs()
+
+        debug('SETUP SOURCES')
+        working_path = os.getcwd() if self.relative_project_dir in os.path.join(os.getcwd(), '') else os.path.join(os.getcwd(), self.relative_project_dir)
+        debug("working_path: %s" % working_path)
+
+        for directory, unused, filenames in os.walk(working_path):
+            debug('WALK')
+            dir_artifacts_with_path = [os.path.join(directory, f) for f in filenames]
+
+            debug(dir_artifacts_with_path)
+            for artifact in dir_artifacts_with_path:
+                debug("  Copying source file %s" % artifact)
+                if os.path.isfile(artifact):
+                    shutil.copy(artifact, self.rpmbuild_gitcopy)
+                    shutil.copy(artifact, self.rpmbuild_sourcedir)
+
+        # NOTE: The spec file we actually use is the one exported by git
+        # archive into the temp build directory. This is done so we can
+        # modify the version/release on the fly when building test rpms
+        # that use a git SHA1 for their version.
+        self.spec_file_name = os.path.basename(find_spec_like_file(self.rpmbuild_sourcedir))
+        self.spec_file = os.path.join(
+            self.rpmbuild_sourcedir, self.spec_file_name)
+
+        self.old_cwd = os.getcwd()
+        if self.relative_project_dir not in os.path.join(os.getcwd(), ''):
+            os.chdir(os.path.join(self.old_cwd, self.relative_project_dir))
+
+        # NOTE: 'which' may not be installed... (docker containers)
+        (status, output) = getstatusoutput("which git-annex")
+        if status != 0:
+            msg = "Please run '%s' as root." % self.package_manager.install(["git-annex"])
+            error_out('%s' % msg)
+
+        run_command("git-annex lock")
+        annexed_files = run_command("git-annex find --include='*'").splitlines()
+        run_command("git-annex get")
+        run_command("git-annex unlock")
+        debug("  Annex files: %s" % annexed_files)
+
+        for annex in annexed_files:
+            debug("Copying unlocked file %s" % annex)
+            if os.path.isfile(os.path.join(self.rpmbuild_gitcopy, annex)):
+                os.remove(os.path.join(self.rpmbuild_gitcopy, annex))
+            shutil.copy(annex, self.rpmbuild_gitcopy)
+
+        self._lock()
+        os.chdir(self.old_cwd)
+
+    def tgz(self):
+        """ Override parent behavior, we already have a tgz. """
+        # TODO: Does it make sense to allow user to create a tgz for this type
+        # of project?
+        self._setup_sources()
+        self.ran_tgz = True
+
+        debug("Scanning for sources.")
+        cmd = "/usr/bin/spectool --list-files '%s' | awk '{print $2}' |xargs -l1 --no-run-if-empty basename " % self.spec_file
+        result = run_command(cmd)
+        if result:
+            self.sources = []
```

Comment:
I don't know if run_command strips the final newline, but this can use a list comprehension:

```python
self.sources = [os.path.join(self.rpmbuild_gitcopy, source) for source in result.split("\n")]
```

You can also consider not using the xargs and use ```os.basename(source)``` in there.